### PR TITLE
Add tabs to code examples with object format.

### LINKED ISSUE #1674: Code examples with tabs
Let's have some api to create code examples with tabs on top of it.

Here is a real example:
image

Let's change it to
```ts
block.code({
   'npm': 'npm install @vuestic/agGrid-theme',
   'yarn': 'yarn add @vuestic/agGrid-theme'
})
```
I'm talking about this kind of tabs:
image

### PR DESCRIPTION
**Pull request by xiongmao86**
## Description
Add tabs to code examples.
fixes #1674

## Markup:
Pass in code object with keys as tab, values as code content.
<details>
```vue
block.code({
   'npm': 'npm install @vuestic/agGrid-theme',
   'yarn': 'yarn add @vuestic/agGrid-theme'
})
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/docs/src/components/DocsCode.vue [hunk 1]
```diff
@@ -16,10 +26,36 @@ export default defineComponent({
...
     language: { type: String as PropType<string>, default: 'javascript' },
-    code: { type: String as PropType<string>, default: '' },
+    code: { type: [Object, String] as PropType<Record<string, string> | string>, default: '' },
   },
   setup (props) {
-    const formattedCode = computed(() => applyTranslations(props.code.replace(/^\n|\n$/g, '')))
+    const isString = computed(() => typeof props.code === 'string')
+
+    const tabs = computed(() => {
+      const result: Array<string> = []
```
Comment (asvae) [inline packages/docs/src/components/DocsCode.vue:35]: You can replace `Array<string>` with `string[]`. Does the same, looks better :).

**Comment by asvae** [comment]
Thanks!

Can we apply this docs feature for some of the docs as well?

The immediate problem I see is that docs configs declaration doesn't have tabs defined in types. So right now we would have to immediately follow your effort up, so that it doesn't become dead code. :)

Makes sense?

File: packages/docs/src/components/DocsCode.vue (hunk 1 above)
Comment (xiongmao86) [reply packages/docs/src/components/DocsCode.vue:35]: Absolutely.

File: packages/docs/src/components/DocsCode.vue [hunk 2]
```diff
@@ -16,10 +26,36 @@ export default defineComponent({
...
+      if (isString.value) {
+        return result
+      } else {
+        for (const key of Object.keys(props.code)) {
+          result.push(key)
+        }
+        return result
+      }
+    })
```
Comment (RVitaly1978) [inline packages/docs/src/components/DocsCode.vue:44]:
```suggestion
    const tabs = computed(() => isString.value ? [] : Object.keys(props.code))
```

File: packages/docs/src/components/DocsCode.vue [hunk 3]
```diff
@@ -16,10 +26,36 @@ export default defineComponent({
...
+      const result: Array<string> = []
+      if (isString.value) {
+        result.push(applyTranslations(props.code.replace(/^\n|\n$/g, '')))
+      } else {
+        for (const key of tabs.value) {
+          result.push(applyTranslations(props.code[key].replace(/^\n|\n$/g, '')))
+        }
+      }
+      return result
+    })
```
Comment (RVitaly1978) [inline packages/docs/src/components/DocsCode.vue:56]:
```suggestion
    const contents = computed(() => isString.value
      ? [applyTranslations(props.code.trim())]
      : tabs.value.map(tab => applyTranslations(props.code[tab].trim()))
    )
```

File: packages/docs/src/components/DocsCode.vue [hunk 4]
```diff
@@ -16,10 +26,36 @@ export default defineComponent({
...
+
+    const index = ref(1)
```
Comment (RVitaly1978) [inline packages/docs/src/components/DocsCode.vue:58]:
```suggestion
    const index = ref(0)
```

File: packages/docs/src/components/DocsCode.vue [hunk 5]
```diff
@@ -1,6 +1,16 @@
...
+        v-for="tab in tabs"
+        :key="tab"
+      >
+        {{ tab }}
+      </va-tab>
+    </template>
+  </va-tabs>
   <prism-wrapper
-    :code="formattedCode"
+    :code="contents[index-1]"
```
Comment (RVitaly1978) [inline packages/docs/src/components/DocsCode.vue:13]:
```suggestion
    :code="contents[index]"
```

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]

**Comment by xiongmao86** [comment]
> Thanks!
> 
> Can we apply this docs feature for some of the docs as well?
> 
> The immediate problem I see is that docs configs declaration doesn't have tabs defined in types. So right now we would have to immediately follow your effort up, so that it doesn't become dead code. :)
> 
> Makes sense?

Ah, I see it now. Thanks.

File: packages/docs/src/components/DocsCode.vue (hunk 5 above)
Comment (xiongmao86) [reply packages/docs/src/components/DocsCode.vue:13]: `model:value` of `va-tabs` start with 1, but contents start with 0, I have deliberately checked this out in demo book that start with `yarn serve`, so I think `index` should probably stay the same.

Comment (RVitaly1978) [reply packages/docs/src/components/DocsCode.vue:13]: I guess there is a bug in our `VaTabs` component with tab indexes. So, I opened a PR #1702 . Let's wait for his approval.

Comment (xiongmao86) [reply packages/docs/src/components/DocsCode.vue:13]: Sure, thank you @RVitaly1978.

Comment (RVitaly1978) [reply packages/docs/src/components/DocsCode.vue:13]: PR https://github.com/epicmaxco/vuestic-ui/pull/1702 has been approved and merged into the develop branch. So now `model:value` of `va-tabs` is equal to its index. You can apply this suggestion)

File: packages/docs/src/components/DocsCode.vue (hunk 4 above)
Comment (RVitaly1978) [reply packages/docs/src/components/DocsCode.vue:41]: see as above

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]

**Comment by xiongmao86** [comment]
This is ready for review.

**Review by m0ksem — APPROVED** [review]
Thanks! 🎉

Checked with  https://github.com/epicmaxco/vuestic-ui/pull/1702.

File: packages/docs/src/page-configs/extensions/ag-grid/page-config.ts [hunk 6]
```diff
@@ -28,7 +28,7 @@ const config: ApiDocsBlock[] = [
   block.link('ag-grid.otherTables.dataTable', '/ui-elements/data-table'),
   block.headline('ag-grid.installation'),
   block.paragraph('ag-grid.dependencies'),
-  block.code(installCommand, 'bash'),
+  block.code(installCommandObject, 'bash'),
```
Comment (xiongmao86) [inline packages/docs/src/page-configs/extensions/ag-grid/page-config.ts:31]: I think I have update [icon-fonts](https://vuestic.dev/en/getting-started/configuration-guide#icon-fonts) here.

File: packages/docs/src/page-configs/extensions/ag-grid/page-config.ts [hunk 7]
```diff
@@ -1,10 +1,10 @@
 import { ApiDocsBlock, Dependencies } from '@/types/configTypes'
 import { PageGenerationHelper } from '@/helpers/DocsHelper'
 
-const installCommand = `
-npm install @vuestic/ag-grid-theme
-// $t('all.code.or')
-yarn add @vuestic/ag-grid-theme`
+const installCommandObject = {
```
Comment (xiongmao86) [inline packages/docs/src/page-configs/extensions/ag-grid/page-config.ts:4]: I think I have update [ag-grid#ag-grid-installation](https://vuestic.dev/en/extensions/ag-grid#ag-grid-installation) here.

File: packages/docs/src/page-configs/getting-started/configuration-guide/code-examples/icons.ts [hunk 8]
```diff
@@ -1,8 +1,7 @@
-export const iconsInstall = `
-yarn add material-design-icons-iconfont -D
-// {{ $t('all.code.or') }}
-npm install material-design-icons-iconfont -D
-`
+export const iconsInstallObject = {
```
Comment (xiongmao86) [inline packages/docs/src/page-configs/getting-started/configuration-guide/code-examples/icons.ts:1]: And update [configuration-guide#icons-fonts](https://vuestic.dev/en/getting-started/configuration-guide#icon-fonts)

**Comment by xiongmao86** [comment]
Thank you for helping me with this pr, @asvae @RVitaly1978 @m0ksem.

**Pull request merged**